implementations/rust/ockam/ockam_transport_tcp/src/workers/receiver.rs

```diff
@@ -132,24 +131,24 @@ impl Processor for TcpRecvProcessor {
         let protocol_version = match self.read_half.read_u8().await {
             Ok(p) => p,
             Err(e) => {
+                trace!("Cannot read the Ockam protocol version: {:?}", &e);
                 self.notify_sender_stream_dropped(ctx, e).await?;
-                return Err(TransportError::GenericIo)?;
+                self.registry
```

Comment:
No need to do it here, it'll be done in the shutdown. But we need to stop the processor using ctx.stop_adress